src/multi-selector/MultiSelector.jsx

```diff
@@ -49,28 +52,44 @@ class MultiSelector extends React.Component {
         };
     }
 
-    assignItems = async values => {
-        const selected = this.state.selected.concat(values);
-        this.setState({ selected });
-        this.props.onChange(selected);
+    getSelected = () => {
+        return this.isControlled ? this.props.selected : this.state.selected;
+    };
+
+    updateSelected = updateFn => {
+        const selected = this.getSelected();
+        const selectedUpdated = updateFn(selected);
+
+        if (this.isControlled) {
+            this.props.onChange(selectedUpdated);
+        } else {
+            this.setState({ selected: selectedUpdated });
+            this.props.onChange(selectedUpdated);
+        }
+
+        // <GroupEditor> requires the props assign/unassign/orderChange to return a promise
+        return Promise.resolve();
```

Comment:
Adding async to the method already returns a promise implicitly and makes the code simpler

Reply:
I'd prefer to be explicit. I had written this code only one month ago and had already forgotten about this, so today I removed the `async` on my refactor, because apparently it's not needed, and then you get a very cryptic error. Then I remembered why we had the async :)

Reply:
Ah, okay. No worries then, re-adding it!